File: src/main/java/org/cerion/stocklist/repository/PriceListRepository.java
package org.cerion.stocklist.repository;

import org.cerion.stocklist.Price;
import org.cerion.stocklist.PriceList;
import org.cerion.stocklist.model.HistoricalDates;
import org.cerion.stocklist.model.Interval;

import java.util.List;

public interface PriceListRepository {

    void add(PriceList list);
    HistoricalDates getHistoricalDates(String symbol, Interval interval);

    @Deprecated
    PriceList get(String symbol, Interval interval, int max);

    List<Price> get(String symbol, Interval interval);
    void deleteAll();
}

File: src/main/kotlin/repository/PriceListRepository.kt
package org.cerion.stocks.core.repository

import org.cerion.stocks.core.PriceList
import org.cerion.stocks.core.PriceRow
import org.cerion.stocks.core.model.HistoricalDates
import org.cerion.stocks.core.model.Interval

interface PriceListRepository {
    fun add(list: PriceList)

    fun getHistoricalDates(symbol: String, interval: Interval): HistoricalDates?
    fun get(symbol: String, interval: Interval): List<PriceRow>
    fun deleteAll()
    
    @Deprecated("")
    operator fun get(symbol: String, interval: Interval, max: Int): PriceList
}
